Common/Core/RecoDecay.h

```diff
@@ -718,13 +728,19 @@ class RecoDecay
         }
         // Printf("MC Rec: Good mother: %d", indexMother);
         auto particleMother = particlesMC.rawIteratorAt(indexMother - particlesMC.offset());
+        dauCounter = 0; // reset daughter counter for the new particleMother
         // Check the daughter indices.
         if (!particleMother.has_daughters()) {
           // Printf("MC Rec: Rejected: bad daughter index range: %d-%d", particleMother.daughtersIds().front(), particleMother.daughtersIds().back());
           return -1;
         }
-        // Check that the number of direct daughters is not larger than the number of expected final daughters.
-        if (particleMother.daughtersIds().back() - particleMother.daughtersIds().front() + 1 > static_cast<int>(N)) {
+        // Check that the number of direct daughters coming from the dacay is not larger than the number of expected final daughters.
+        for (const auto& dau : particleMother.template daughters_as<o2::aod::McParticles>()) {
```

Comment:
```suggestion
        for (const auto& dau : particleMother.template daughters_as<T>()) {
```
We should absolutely avoid hard-coding any table types.
Is the casting even needed?

Reply:
Replacing the simple index difference with looping over the actual particles will slow this check down significantly. Did you do any CPU time checks?

Reply:
I will remove the hard-coded table.
I need to work with elements of the McParticles table in order to be able to retrieve the information about the production process of each daughter (and the check on the production process needs to be done for all the daughters), while the function daughtersIds would return the daughter tracks slice.
I did not do CPU time checks, but I would also not know how to implement this differently.